src/main/java/org/cactoos/io/GzipInput.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.GZIPInputStream;
+import org.cactoos.Input;
+
+/**
+ * Input that reads from a Gzip content.
+ *
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.29
+ */
+public final class GzipInput implements Input {
+
+    /**
+     * The input.
+     */
+    private final Input origin;
+
+    /**
+     * The buffer size.
+     */
+    private final int size;
+
+    /**
+     * Ctor.
+     * @param input The input.
+     */
+    public GzipInput(final Input input) {
+        // @checkstyle MagicNumberCheck (1 line)
+        this(input, 16 << 10);
```

Comment:
@fabriciofx What does ``16 << 10`` mean? Could you write a plain ``int`` so it will be easier to understand?

Reply:
@amihaiemil It's the buffer's size. I just put the same value used by Yegor [here](https://github.com/yegor256/cactoos/blob/master/src/main/java/org/cactoos/io/InputAsBytes.java#L61).

Reply:
@fabriciofx I see, but I would still go with an integer -- what does ``16 << 10`` evaluate to, I don't know, and I bet I am not the only one :D.  ``<<`` is one of the strangest operators in Java, I have never used it in 4 years so far

Reply:
@amihaiemil As far as I know, `2 << n` is a super quick way to rise `2` to the power of `n`. `16` is a power of `2`: `16 == 2⁴`. So `16 << 10 == 2¹⁴` (check it). Programmers have an old tradition of allocating buffers with the size of the power of `2`. I don't know why. Probably somehow related to the memory alignment or something. :)